src/main/java/jenkins/plugins/purgejobhistory/PurgeJobHistory.java

```diff
@@ -85,9 +107,12 @@ public static void purge(Job<?, ?> job, boolean resetNextBuildNumber) throws IOE
                 acl.checkPermission(Run.DELETE);
                 lastACL = acl;
             }
+            if (!force && run.isKeepLog()) {
+                continue;
+            }
             run.delete();
         }
-        if (resetNextBuildNumber) {
+        if (resetNextBuildNumber && job.getLastBuild() == null) {
             job.updateNextBuildNumber(1);
```

Comment:
Should probably reset to `latest build number + 1` if there are builds left, right?